### PR TITLE
extend public API with optional event functions to support additional use-case

### PR DESCRIPTION
**Pull request by weisjohn**
Hey @legomushroom, thanks for making this.

I added these four optional functions so that I could synchronize an animation over-top a `<video>` element. That is, when I integrate with these elements, I can pause, seek, and play the video, scrub back and forth, etc.

This is done as:

```javascript
const vid = document.querySelector('#vid');

// simple state holder for play/pause
const state = {};

// connect to player
const player = new MojsPlayer({
  add: timeline,
  isPlaying: true,
  isRepeat: true,
  isSaveState: false,
  progress: 0,
  onPlayStateChange: isPlaying => {
    state.isPlaying = isPlaying;
    isPlaying ? vid.play() : vid.pause();
  },
  onSeekStart: () => {
    vid.pause();
    state.seeking = true;
  },
  onSeekEnd: () => {
    if (state.isPlaying) vid.play();
    state.seeking = false;
  },
  onProgress: progress => {
    if (!state.seeking) return;

    // set video time based on scrubber
    vid.currentTime = vid.duration * progress;
  }
});
```

I can DM a gif/vid of the use case if you want to see a justification of the usefulness of these additions.

Thanks for considering.

### CODE REVIEW

**Comment by legomushroom** [comment]
Hey @weisjohn ! Thanks a lot, sounds nice, will look into it just shortly. 👍

File: js/mojs-player.babel.js [hunk 1]
```diff
@@ -244,7 +248,14 @@ class MojsPlayer extends Module {
     @private
     @param {Object} Original event object.
   */
-  _onSeekStart ( e ) { this._sysTween.pause(); }
+  _onSeekStart ( e ) {
+    this._sysTween.pause();
+
+    const { onSeekStart } = this._props;
+    if (onSeekStart) {
```
Comment (legomushroom) [inline js/mojs-player.babel.js]: Let's be sure an actual `function type` was passed as callback here, we can have `_isFunction` helper and use it for all 4 examples like this:

```javascript
  const { onSeekStart } = this._props;
  if (this._isFunction(onSeekStart)) {
    // run
  }
```

The `_isFunction` can be as simple as:
```javascript
  // ...
  _isFunction (testVariable) {
    return typeof testVariable === 'function';
  }
  // ...
```

**Review by legomushroom — CHANGES_REQUESTED** [review]
Just one nit, otherwise good.

**Comment by weisjohn** [comment]
@legomushroom I just made those changes. I haven't used the "Changes requested" experience on GH yet: do I need to submit a new PR or were those additions enough?

**Comment by legomushroom** [comment]
@weisjohn thanks! No need for separate `PR` -  I see your commits here. Will review just in few moments.

**Comment by legomushroom** [comment]
Looks good, thanks a lot. 👍

**Comment by legomushroom** [comment]
If you want to contribute more, you can think how to provide the callbacks with state variables, something like this:

```javascript
const vid = document.querySelector('#vid');

// connect to player
const player = new MojsPlayer({
  add: timeline,
  isPlaying: true,
  isRepeat: true,
  isSaveState: false,
  progress: 0,
  onPlayStateChange: isPlaying => {
    isPlaying ? vid.play() : vid.pause();
  },
  onSeekStart: (isPlaying) => {
    vid.pause();
  },
  onSeekEnd: (isPlaying) => {
    if (state.isPlaying) vid.play();
  },
  onProgress: progress, isSeeking => {
    if (!isSeeking) return;

    // set video time based on scrubber
    vid.currentTime = vid.duration * progress;
  }
});
```

**Pull request merged**